src/main/java/edu/kis/vh/nursery/defaultCountingOutRhymer.java

```diff
@@ -1,34 +1,45 @@
 package edu.kis.vh.nursery;
+import edu.kis.vh.nursery.collection.IntArrayStack;
+import edu.kis.vh.nursery.collection.IntLinkedList;
+import edu.kis.vh.nursery.collection.StackImplementation;
 
 public class defaultCountingOutRhymer {
-
-	private int[] NUMBERS = new int[12];
-
-	public int total = -1;
-
-	public void countIn(int in) {
-		if (!isFull())
-			NUMBERS[++total] = in;
+	
+	private StackImplementation stackImplementation;
+	
+	public defaultCountingOutRhymer() {
+		this("array");
 	}
-
-		public boolean callCheck() {
-			return total == -1;
+	public defaultCountingOutRhymer(String string) {

```

Comment:
The decision depends on a string value is a bad practice.  Enum, constant would be better. The best solution would be passing an instance of StackImplementation to a constructor (as a parameter instead of a string).